src/pages/index.js

```diff
@@ -1,122 +1,187 @@
-  },
-];
-//---------------Elements
-const cardTemplate = document.querySelector("#card-template");
-//profile
-const profileEditButton = document.querySelector("#profile-edit-button");
-
-const profileEditForm = document.forms["modal__form"];
-const profileNameInput = profileEditForm.querySelector("#profile-title-input");
-const profileJobInput = profileEditForm.querySelector(
-  "#profile-description-Input"
+const cardSection = new Section(renderCard, constants.cardListEl);
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
 
-//cards add modal
-const cardListEl = document.querySelector(".cards__list");
-const addNewCardButton = document.querySelector(".profile__add-button");
-const cardAddModal = document.querySelector("#add-card-modal");
-const addCardFormElement = cardAddModal.querySelector(".modal__form");
-//cards input
-
-//image modal buttons
-const config = {
-  formSelector: ".modal__form",
-  inputSelector: ".modal__input",
-  submitButtonSelector: ".modal__button",
-  inactiveButtonClass: "modal__button_disabled",
-  inputErrorClass: "modal__input_type_error",
-  errorClass: "modal__error_visible",
-};
-
-const editFormValidator = new FormValidator(config, profileEditForm);
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f1dab6ed-e5ba-44ef-bef1-7369d7e9bf0d",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.setUserAvatar(res);
+  })
+  .catch((err) => {
+    console.error(`Error ${err}`);
+  });
+api
+  .getInitialCards()
+  .then((res) => {
+    cardSection.renderItems(res);
+  })
+  .catch((err) => {
+    console.error(`Error ${err}`);
+  });
+
+const cardDeleteModal = new PopupQuestion("#delete-card-modal");
+cardDeleteModal.setEventListeners();
+const editFormValidator = new FormValidator(
+  constants.config,
+  constants.profileEditForm
+);
 editFormValidator.enableValidation();
-const addFormValidator = new FormValidator(config, addCardFormElement);
-addFormValidator.enableValidation();
 
+const addFormValidator = new FormValidator(
+  constants.config,
+  constants.addCardFormElement
+);
+addFormValidator.enableValidation();
+addFormValidator.toggleButtonState();
+const profileAvatarFormValidator = new FormValidator(
+  constants.config,
+  constants.profileAvatarModalForm
+);
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link) => {
+    cardPopup.renderLoading(true);
+    api
+      .addCard(link)
+
+      .then((res) => {
+        cardSection.addItem(res);
+        cardPopup.close();
+        cardPopup.renderLoading(false);
```

Comment:
`.renderLoading(false);` should be in `finally`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.